apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog.Tests/GoogleStackdriverTargetTest.cs

```diff
@@ -134,16 +148,128 @@ private LogEventInfo CreateLog(LogLevel level, string msg)
+        {
+            var target = new GoogleStackdriverTarget
+            {
+                ProjectId = "a_project_id",
+                SendJsonPayload = true,
+                JsonConvertFunction = o => Value.ForNull(),
+                JsonConvertTypeName = "a_type_name",
+            };
+            var nlogEx = await Assert.ThrowsAsync<NLogRuntimeException>(() => ActivateTargetAsync(target));
+            Assert.IsType<InvalidOperationException>(nlogEx.InnerException);
+        }
+
+        [Fact]
+        public async Task InvalidOptions_BadJsonConverterType()
+        {
+            var target = new GoogleStackdriverTarget
+            {
+                ProjectId = "a_project_id",
+                SendJsonPayload = true,
+                JsonConvertTypeName = "a_type_name",
+                JsonConvertMethodName = "a_method_name",
+            };
+            var nlogEx = await Assert.ThrowsAsync<NLogRuntimeException>(() => ActivateTargetAsync(target));
+            Assert.IsType<InvalidOperationException>(nlogEx.InnerException);
+            Assert.Contains("a_type_name", nlogEx.InnerException.Message);
+        }
+
+        private const string ConverterMsg = "Converter called :)";
+        private class TestJsonConverter
+        {
+            public static ConcurrentBag<string> SeenItems { get; } = new ConcurrentBag<string>();
+
+            public static Value TestJsonConverterMethod_Value_Static(object o)
+            {
+                SeenItems.Add(JToken.FromObject(o).Value<string>());
+                return Value.ForNull();
+            }
+
+            public Value TestJsonConverterMethod_Value_Instance(object o)
+            {
+                SeenItems.Add(JToken.FromObject(o).Value<string>());
+                return Value.ForNull();
+            }
+            public static JToken TestJsonConverterMethod_JToken_Static(object o)
+            {
+                SeenItems.Add(JToken.FromObject(o).Value<string>());
+                return null;
+            }
+
+            public JObject TestJsonConverterMethod_JObject_Instance(object o)
+            {
+                SeenItems.Add(JToken.FromObject(o).Value<string>());
+                return null;
+            }
+        }
+
+        [Fact]
+        public async Task InvalidOptions_BadJsonConverterMethod()
+        {
+            var target = new GoogleStackdriverTarget
+            {
+                ProjectId = "a_project_id",
+                SendJsonPayload = true,
+                JsonConvertTypeName = typeof(TestJsonConverter).AssemblyQualifiedName,
+                JsonConvertMethodName = "a_method_name",
+            };
+            var nlogEx = await Assert.ThrowsAsync<NLogRuntimeException>(() => ActivateTargetAsync(target));
+            Assert.IsType<InvalidOperationException>(nlogEx.InnerException);
+            Assert.Contains("a_method_name", nlogEx.InnerException.Message);
+        }
+
+        [Fact]

```

Comment:
Is there an existing test for the default converter being applied?

Reply:
Yes, `SingleLogEntryWithJsonProperties` and `SingleLogEntryWithJsonCollectionProperties` test the default converter.